### ISSUE TITLE
Version 1.4.4 is not compatible with flutter 1.22.0-12.1.pre

### ISSUE THREAD

**Issue by dluksza** [MaikuB/flutter_local_notifications#809]
After updating flutter to the latest beta version dependency resolution is not able to resolve version of platform dependency between `flutter_local_notifications` and `flutter_driver`. Here is the error output:

```
Flutter is already up to date on channel beta
Flutter 1.22.0-12.1.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision 8b3760638a (4 days ago) • 2020-09-15 17:47:13 -0700
Engine • revision 4654fc6cf6
Tools • Dart 2.10.0 (build 2.10.0-110.3.beta)
Because every version of flutter_driver from sdk depends on platform 3.0.0-nullsafety.1 and flutter_local_notifications >=0.9.0 depends on platform ^2.2.1, flutter_driver from sdk is incompatible with flutter_local_notifications >=0.9.0.
So, because habit_challenge depends on both flutter_local_notifications 1.4.4+4 and flutter_driver any from sdk, version solving failed.
Running "flutter pub upgrade" in HabitChallenge2...                     
pub upgrade failed (1; So, because habit_challenge depends on both flutter_local_notifications 1.4.4+4 and flutter_driver any from sdk, version solving failed.)
```

**To Reproduce**
1. Switch to flutter beta channel
2. Run `flutter pub get`
3. Dependency resolution fails with the above error

**Comment by MaikuB** [comment]
Duplicate of #782 and was discussed in a PR related to it (#785). I would say you need to maintain your own fork for now as I'd rather not have the plugin depend on pre-release dependency

**Comment by dluksza** [comment]
As PR you pointed out was merged-in I'll just depend on the _git_ version for now.

Here's `pubspec.yaml` entry of anyone that would hit the same error

```yaml
  flutter_local_notifications:
    git:
      url: git://github.com/MaikuB/flutter_local_notifications
      ref: 4ec7dfa9694dd2c7f1528ecd4381269bdd216382
      path: flutter_local_notifications
```

**Comment by MaikuB** [comment]
@dluksza in case you missed it, it was reverted

**Comment by ghenry** [comment]
I'm on dev channel testing xcode 12 and had to do this in my pubsec in dev
dependencies.

   platform: 3.0.0-nullsafety.1

Works for me.

**Comment by dluksza** [comment]
@MaikuB I know that it was reverted, therefore I've pinned to the commit, not to a `master` branch. Haven't found any issues with it.

**Comment by MaikuB** [comment]
@dluksza cool, just wanted to make sure :)

**Comment by yanivshaked** [comment]
Official version of Flutter 1.22.0 is out today, and this issue still exists (platform 3.0.0-nullsafety.2 is in use).

**Comment by MaikuB** [comment]
Should be fixed in 1.4.4+5
